server/game/event.js

```diff
@@ -27,8 +27,10 @@ class Event {
             return;
         }
 
-        card.markAsSaved();
-        this.cards = _.reject(this.cards, c => c === card);
+        card.game.raiseMergedEvent('onCharacterSaved', { card: card }, () => {
+            card.markAsSaved();
+            this.cards = _.reject(this.cards, c => c === card);
+        });
 
         if(_.isEmpty(this.cards)) {
```

Comment:
This `if` needs to be inside the event handler, because removing the card is no longer immediate.